packages/block-editor/src/components/rich-text/use-before-input-rules.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef } from '@wordpress/element';
+import { useRefEffect } from '@wordpress/compose';
+import { insert, isCollapsed } from '@wordpress/rich-text';
+import { useDispatch } from '@wordpress/data';
+import { applyFilters } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+
+/**
+ * When typing over a selection, the selection will we wrapped by a matching
+ * character pair. The second character is optional, it defaults to the first
+ * character.
+ *
+ * @type {string[]} Array of character pairs.
+ */
+const wrapSelectionSettings = [
+	'`',
+	'()',
```

Comment:
I'm a bit worried with expanding these to characters that are natural openers because I suspect it's a typical action to select a word and type the first letter of what should replace it — parentheses and quotation marks, in particular, would have a much higher incidence of cases where the intention is indeed to replace the selected text with the mark, not to wrap it in them.

Reply:
Should we try it and see if it's discovered? If it's doing something unexpected, we can always remove some of the characters easily?

Reply:
I'm fine with it either way, just curious how people will react to it.

Reply:
I think it's worth trying to see how it feels in practice, but I think I'd rather add them a bit more incrementally so we can focus feature announcements on the ones we are more certain about first without the ones that could be problematic tainting the feature.

Reply:
Ok, I removed everything but `` ` `` and the quotes. Or should we remove the quotes too?

Reply:
I remember someone asking for the quotes back when we first added [pasting a link over selection](https://core.trac.wordpress.org/ticket/31571), but I can't find that conversation now...